esprima.js

```diff
@@ -1898,6 +1912,10 @@
         finishProgram: function (body) {
             this.type = Syntax.Program;
             this.body = body;
+            if (sourceType === 'module') {
+                // very restrictive for now
```

Comment:
https://github.com/estree/estree/pull/59#issuecomment-86506047

Please update the PR so it always includes sourceType.